### ISSUE TITLE
Why dice loss is -dice coef?

### ISSUE THREAD

**Issue by godissaw** [ellisdg/3DUnetCNN#20]
Thanks for your new update. But I want to know , why the dice loss seems like this?
`def dice_coef(y_true, y_pred, smooth=1.):
    y_true_f = K.flatten(y_true)
    y_true_onehot = K.one_hot(tf.to_int32(y_true_f),8)
    y_true_onehot_f = K.flatten(y_true_onehot)
    y_pred_f = K.flatten(y_pred)    
    intersection = K.sum(y_true_onehot_f * y_pred_f)
    return (2. * intersection + smooth) / (K.sum(y_true_onehot_f) + K.sum(y_pred_f) + smooth)

def dice_coef_loss(y_true, y_pred):
    return -dice_coef(y_true, y_pred)`
I rewrited your dice loss to train the multi-category task, but I didn't know why this loss is just equal to the 
"-dice_coef"
thanks.

**Comment by ellisdg** [comment]
@godissaw I don't really understand the question. What do you think is wrong with the loss function?

**Comment by godissaw** [comment]
@ellisdg 
Um, no, no problem. I'm sorry because  then I didn't really understand how the dice loss works. sorry to bother you XD.

**Comment by ellisdg** [comment]
No problem! Just in case it is still unclear: The dice coefficient measures the similarity between the prediction and the truth. A higher dice coefficient means more similarity between the prediction and truth, which is what we are looking for. The loss function is the negative of the dice coefficient because the training tries to minimize the loss function. It is set up so that the goal of the CNN during training is to find the lowest negative dice coefficient.
low negative dice coefficient (loss function) <==> high positive dice coefficient <==> good match between prediction and truth
